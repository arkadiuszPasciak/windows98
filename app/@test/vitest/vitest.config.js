import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
