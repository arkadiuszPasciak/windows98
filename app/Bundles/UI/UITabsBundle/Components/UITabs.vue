<template>
  <div class="UITabs">
    <div class="tabs">
      <button
        v-for="(tab, id) in tabs"
        :key="`ui-tabs-button-${tab}-${id}`"
        class="tab"
        :class="id === activeIndex ? 'active' : ''"
        type="button"
        :data-test="`ui-tabs-button-${id}`"
        @click="setActiveTab(id)"
      >
        {{ t(tab.title) }}
      </button>
    </div>
    <div class="content">
      <component
        :is="tab.component"
        v-for="(tab, id) in tabs"
        :key="`ui-tabs-content-${tab}-${id}`"
        class="content-tab"
        :class="id === activeIndex ? 'active' : ''"
      />
    </div>
  </div>
</template>

<script setup lang="ts">
import { PropType, ref } from "vue"
import { useI18n } from "vue-i18n"
import { TUITabsProps } from "@APP|Bundles/UITabsBundle/Supports/UITabs.supports"

defineProps({
	tabs: {
		type: Object as PropType<TUITabsProps>,
		required: true,
	},
})

const { t } = useI18n()

const activeIndex = ref<number>(0)

const setActiveTab = (id: number): void => {
	if (activeIndex.value === id) {
		return
	}

	activeIndex.value = id
}
</script>

<style
  lang="scss"
  scoped
  src="@APP|Bundles/UITabsBundle/Styles/UITabs.styles.scss"
/>
