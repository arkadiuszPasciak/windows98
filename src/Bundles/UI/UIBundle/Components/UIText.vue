<template>
  <p class="UIText">
    <slot />
  </p>
</template>

<script setup lang="ts"></script>

<style lang="scss" scoped src="@Bundles/UIBundle/Styles/UIText.styles.scss" />
