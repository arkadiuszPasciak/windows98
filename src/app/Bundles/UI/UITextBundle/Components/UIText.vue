<template>
  <p class="UIText">
    <slot />
  </p>
</template>

<script setup lang="ts"></script>

<style
  lang="scss"
  scoped
  src="@APP|Bundles/UITextBundle/Styles/UIText.styles.scss"
/>
