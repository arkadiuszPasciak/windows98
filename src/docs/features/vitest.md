# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
