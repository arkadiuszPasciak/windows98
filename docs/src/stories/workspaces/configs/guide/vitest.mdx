import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

