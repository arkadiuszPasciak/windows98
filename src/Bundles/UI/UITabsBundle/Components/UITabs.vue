<template>
  <div class="UITabs">
    <div class="tabs">
      <button
        v-for="(tab, id) in tabs"
        :key="`ui-tabs-button-${tab}-${id}`"
        class="tab"
        :class="id === activeIndex ? 'active' : ''"
        type="button"
        :data-test="`ui-tabs-button-${id}`"
        @click="setActiveTab(id)"
      >
        {{ t(tab.title) }}
      </button>
    </div>
    <div class="content">
      <component
        :is="tab.component"
        v-for="(tab, id) in tabs"
        :key="`ui-tabs-content-${tab}-${id}`"
        class="content-tab"
        :class="id === activeIndex ? 'active' : ''"
      />
    </div>
  </div>
</template>

<script setup lang="ts">
  import { PropType, Ref, ref } from 'vue'
  import { useI18n } from 'vue-i18n'
  import { TUITabsProps } from '@Bundles/UITabsBundle/Supports/UITabs.supports'

  defineProps({
    tabs: {
      type: Object as PropType<TUITabsProps>,
      required: true,
    },
  })

  const { t } = useI18n()

  const activeIndex = ref(0) as Ref<number>

  const setActiveTab = (id: number): void => {
    if (activeIndex.value === id) {
      return
    }

    activeIndex.value = id
  }
</script>

<i18n src="@Bundles/SettingsBundle/Locales/Settings.locales.json"></i18n>

<style
  lang="scss"
  scoped
  src="@Bundles/UITabsBundle/Styles/UITabs.styles.scss"
/>
