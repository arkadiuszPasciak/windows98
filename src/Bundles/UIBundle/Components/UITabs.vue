<template>
  <div class="UITabs">
    <div class="tabs">
      <button
        v-for="(tab, id) in tabs"
        :key="`ui-tabs-button-${tab}-${id}`"
        class="tab"
        type="button"
      >
        {{ tab.title }}
      </button>
    </div>
    <div class="contents">
      <div
        v-for="(tab, id) in tabs"
        :key="`ui-tabs-content-${tab}-${id}`"
        class="content"
      >
        <component :is="tab.component"></component>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
  import { PropType } from 'vue'
  import { TUITabsProps } from '@Bundles/UIBundle/Supports/UITabs.supports'

  defineProps({
    tabs: {
      type: Object as PropType<TUITabsProps>,
      required: true,
    },
  })
</script>

<style lang="scss" scoped>
  @import '@Bundles/UIBundle/Styles/UITabs.styles.scss';
</style>
