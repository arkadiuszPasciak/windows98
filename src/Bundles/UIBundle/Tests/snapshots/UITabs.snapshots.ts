import { shallowMount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import UITabs from '@Bundles/UIBundle/Components/UITabs.vue'
import UIText from '@Bundles/UIBundle/Components/UIText.vue'
import { TUITabsProps } from '@Bundles/UIBundle/Supports/UITabs.supports'

const tabs = [
  {
    title: 'General',
    component: UIText,
  },
  {
    title: 'Display',
    component: UIText,
  },
] as TUITabsProps

const defaultWrapper = shallowMount(UITabs, {
  props: {
    tabs: tabs,
  },
})

describe('[UIBundle]<Snapshots>(UITabs)', async () => {
  it('renders correctly component', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
})
