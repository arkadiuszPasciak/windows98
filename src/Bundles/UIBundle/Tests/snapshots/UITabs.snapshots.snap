// Vitest Snapshot v1

exports[`[UIBundle]<Snapshots>(UITabs) > renders correctly component 1`] = `
<div
  class="UITabs"
  data-v-39833990=""
>
  <div
    class="tabs"
    data-v-39833990=""
  >
    
    <button
      class="tab active"
      data-test="ui-tabs-button-0"
      data-v-39833990=""
      type="button"
    >
      General
    </button>
    <button
      class="tab"
      data-test="ui-tabs-button-1"
      data-v-39833990=""
      type="button"
    >
      General
    </button>
    
  </div>
  <div
    class="content"
    data-v-39833990=""
  >
    
    <anonymous-stub
      class="content-tab active"
      data-v-39833990=""
    />
    <anonymous-stub
      class="content-tab"
      data-v-39833990=""
    />
    
  </div>
</div>
`;
